### PR TITLE
How to use brotli compression algorithm method

### LINKED ISSUE #5731: [API] How to use Brotli compression method for web applications
### Topic Suggestion
How to use Brotli compression method for web applications

### Writing sample(s): 
https://www.section.io/engineering-education/how-to-troubleshoot-and-monitor-a-database/
https://www.section.io/engineering-education/integration-of-ros-with-arduino/

#### Proposed title of article
How to use Brotli compression method for web applications

#### Proposed article introduction
Static assets such as **.js** ,**.CSS** and **.json** text files are commonly used in web application projects, with redundancy and compression playing an important role. Compression reduces the amount of bandwidth required for website rendering, which speeds up the process.
Brotli is a lossless compression algorithm that combines a current form of the LZ77 algorithm, Huffman coding, and 2nd order context modeling to compress data. 

#### Key takeaways 
1. what is Brotli
2. How brotli works
3. Application of brotli
4. Configuring with Nginx
5. Configuring with AWS CloudFront distribution

#### Article quality
This article will add an instance of brotli compression to an already existing instance of Nginx, while also taking the reader through a step by step process in creating a dynamic brotli module.

#### References
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by Newton-cyber**
#### Formatting and structure checks
- [X] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [X] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [X] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [X] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [X] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [X] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [X] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [X] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [X] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [X] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [X] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [X] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [X] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [X] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [X] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes
`This closes #5731`

### CODE REVIEW

**Comment by Newton-cyber** [comment]
@srishilesh Done a new PR but the merge branch keeps on coming to my PR

File: content/articles/how-to-use-brotli-compression-method/index.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+### Introduction
+Static assets such as *.js*, *.CSS* and *.json* text files are commonly used in web application projects, with redundancy and compression playing an important role. ,Data compression is the process of encoding files to reduce their size, making them take up less space on your server and allowing them to be transferred to site clients more quickly. Compression reduces the amount of bandwith required for website rendering, which speeds up the process.
```
Comment (srishilesh) [inline content/articles/how-to-use-brotli-compression-method/index.md:2]:
```suggestion
Static assets such as *.js*, *.css* and *.json* text files are commonly used in web application projects, with redundancy and compression playing an important role.  Data compression is the process of encoding files to reduce their size, making them take up less space on your server and allowing them to be transferred to site clients more quickly. Compression reduces the amount of bandwith required for website rendering, which speeds up the process.

```

File: content/articles/how-to-use-brotli-compression-method/index.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
+### Introduction
+Static assets such as *.js*, *.CSS* and *.json* text files are commonly used in web application projects, with redundancy and compression playing an important role. ,Data compression is the process of encoding files to reduce their size, making them take up less space on your server and allowing them to be transferred to site clients more quickly. Compression reduces the amount of bandwith required for website rendering, which speeds up the process.
+
+### Prerequisites
+- Compression algorithm
```
Comment (srishilesh) [inline content/articles/how-to-use-brotli-compression-method/index.md:5]: Make this as a complete sentence.

File: content/articles/how-to-use-brotli-compression-method/index.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
+### Introduction
+Static assets such as *.js*, *.CSS* and *.json* text files are commonly used in web application projects, with redundancy and compression playing an important role. ,Data compression is the process of encoding files to reduce their size, making them take up less space on your server and allowing them to be transferred to site clients more quickly. Compression reduces the amount of bandwith required for website rendering, which speeds up the process.
+
+### Prerequisites
+- Compression algorithm
+
+### What is Brotli
+Brotli is a lossless data compression algorithm that employs a variant of the *LZ77 algorithm*, as well as *Huffman coding* and *second-order context modelling*. Its compression ratio is comparable to the best general-purpose compression methods currently available.
```
Comment (srishilesh) [inline content/articles/how-to-use-brotli-compression-method/index.md:8]: Please explain these terminologies. Beginners might not understand them easily.

File: content/articles/how-to-use-brotli-compression-method/index.md (hunk 2 above)
Comment (srishilesh) [reply content/articles/how-to-use-brotli-compression-method/index.md:5]: Before jumping into brotli, start with explaining what compression algo is

File: content/articles/how-to-use-brotli-compression-method/index.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+### Prerequisites
+- Compression algorithm
+
+### What is Brotli
+Brotli is a lossless data compression algorithm that employs a variant of the *LZ77 algorithm*, as well as *Huffman coding* and *second-order context modelling*. Its compression ratio is comparable to the best general-purpose compression methods currently available.
+
+### How Brotli works
+The browser sends an `Accept-Encoding` header with the algorithms it supports and their priority order; the server chooses one, uses it to compress the body of the `content-encoding` header to inform the browser of its choice. Because content negotiation was used to select a representation based on its encoding, the server must include a vary header in the response that includes at least `Accept-encoding`; this allows caches to cache the various representations of the resource.
```
Comment (srishilesh) [inline content/articles/how-to-use-brotli-compression-method/index.md:11]: `Accept-Encoding` and `content-encoding`

What is this for? Add example image or sample header to explain them first.

Comment (srishilesh) [reply content/articles/how-to-use-brotli-compression-method/index.md:11]: ` it supports and their priority order;`

What do you mean by priority order?

Comment (srishilesh) [reply content/articles/how-to-use-brotli-compression-method/index.md:11]: `content negotiation was used to select a representation based on its encoding`

Explain

Comment (srishilesh) [reply content/articles/how-to-use-brotli-compression-method/index.md:11]: `caches to cache the various representations of the resource.`

Explain

File: content/articles/how-to-use-brotli-compression-method/index.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+### Application of Brotli
+There are two ways by which we can deliver Brotli compressed assets:
+1. Enabling Brotli on web-server
+2. Enabling Brotli on CDNs
```
Comment (srishilesh) [inline content/articles/how-to-use-brotli-compression-method/index.md:16]: What is a webserver and what is CDN?

File: content/articles/how-to-use-brotli-compression-method/index.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+### Compression with Nginx statically
```
Comment (srishilesh) [inline content/articles/how-to-use-brotli-compression-method/index.md:18]: What is nginx

File: content/articles/how-to-use-brotli-compression-method/index.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    return gulp.src(src)
+        .pipe(brotli.compress({
+            extension: "br",
+            quality: 11
+        }))
+        .pipe(gulp.dest(dest));
+};
+
+exports.brotlicompress = brotlicompress;
+```
```
Comment (srishilesh) [inline content/articles/how-to-use-brotli-compression-method/index.md:37]: Explain

Comment (srishilesh) [reply content/articles/how-to-use-brotli-compression-method/index.md:37]: Where do you store this? Or how will you work with this?

File: content/articles/how-to-use-brotli-compression-method/index.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+</files>
+<files *.svg.br>
+    AddType "image/svg+xml".br
+    AddEncoding br.br
+</files>
+<files *.html.br>
+    AddType "text/html".br
+    AddEncoding br.br
+</files>
+```
```
Comment (srishilesh) [inline content/articles/how-to-use-brotli-compression-method/index.md:61]: Explain

**Review by srishilesh — CHANGES_REQUESTED** [review]
Hey @Newton-cyber ,

On having a quick glance at your article, it looks like we may have close this PR, since it's not very beginner friendly.

Your article:
- Lacks explanation of concepts and terminologies
- The code snippets are not explained properly
- You do not direct the reader with step by step implementation
- Few code snippets looks very technical - you don't specify where or how to run them
- You have covered several topics in one short article. Cover 1 - 2 concepts at max in an article.
- Overall content looks quite low.
- This PR has an unwanted additional image file `integration-of-ros-with-arduino/WhatsApp Image 2021-08-08 at 17.57.26.jpeg`

I'll be closing this PR now. You may work on all the changes and submit only ready to be published PRs.

I highly recommend you to go through approved PRs/Published articles to see how other authors have formatted and structured the flow. This will help you avoid potential PR closure.

Thanks.

**Comment by srishilesh** [comment]
@Newton-cyber 

Do we have any updates?

**Pull request closed without merging**